tests/testthat/test-object_name_linter.R

```diff
@@ -155,3 +155,88 @@ test_that("object_name_linter won't fail if an imported namespace is unavailable
     3L
   )
 })
+
+test_that("object_name_linter supports custom regexes", {
+  # disables default styles
+  linter <- object_name_linter(
+    regexes = c("shinyModule" = rex(start, lower, zero_or_more(alnum), "UI" %or% "Server", end))
+  )
+  msg <- rex::rex("Variable and function name style should match shinyModule.")
+  linter2 <- object_name_linter(
+    styles = c("snake_case", "symbols"),
+    regexes = c("shinyModule" = rex(start, lower, zero_or_more(alnum), "UI" %or% "Server", end))
+  )
+  msg2 <- rex::rex("Variable and function name style should match snake_case, symbols or shinyModule.")
+
+  # Can't allow 0 styles
+  expect_error(
+    object_name_linter(NULL),
+    rex::rex("At least one style must be specified using `styles` or `regexes`.")
+  )
+
+  expect_lint(
+    trim_some('
+      snake_case <- 42L
+      "%+%" <- function(...) ..1 + ..2
+
+      myModuleUI <- function(id) {
+        # blah
+      }
+
+      myModuleServer <- function(id) {
+        # blah
+      }
+
+      myBadName <- 20L
+    '),
+    list(
+      list(line_number = 1L, message = msg),
+      list(line_number = 2L, message = msg),
+      # argument "id" is linted if we only allow shinyModule names
+      list(line_number = 4L, column_number = 24L, message = msg),
+      list(line_number = 8L, column_number = 28L, message = msg),
+      list(line_number = 12L, message = msg)
+    ),
+    linter
+  )
+
+  expect_lint(
+    trim_some('
+      snake_case <- 42L
+      "%+%" <- function(...) ..1 + ..2
+
+      myModuleUI <- function(id) {
+        # blah
+      }
+
+      myModuleServer <- function(id) {
+        # blah
+      }
+
+      myBadName <- 20L
+    '),
+    list(line_number = 12L, message = msg2),
+    linter2
+  )
+
+  # Default regex naming works
+  expect_lint(
+    trim_some("
+      a <- 42L
+      b <- 1L
+      c <- 2L
+    "),
+    list(line_number = 3L, message = rex::rex("Variable and function name style should match /^a$/ or /^b$/.")),
```

Comment:
looks great! I do wonder if `/regex/` is widely understood, but that can easily be fixed later if needed.

Reply:
Yeah, I had the same question, but thought it was clearer than using any other form of quoting or not quoting it at all.